src/server/multi_command_squasher.cc

```diff
@@ -238,6 +274,7 @@ bool MultiCommandSquasher::ExecuteSquashed(facade::RedisReplyBuilder* rb) {
     aborted |= error_abort_ && CapturingReplyBuilder::TryExtractError(replies.back());
 
     CapturingReplyBuilder::Apply(std::move(replies.back()), rb);
+    current_reply_size_.fetch_sub(Size(replies.back()), std::memory_order_relaxed);
```

Comment:
there is std::move line above on replies.back()

Reply:
That move does nothing there, unfortunately. But I've added a fix https://github.com/dragonflydb/dragonfly/pull/4149/files